pkg/controller/controller_utils.go

```diff
@@ -514,14 +516,29 @@ func validateControllerRef(controllerRef *metav1.OwnerReference) error {
 }
 
 func (r RealPodControl) CreatePods(namespace string, template *v1.PodTemplateSpec, object runtime.Object) error {
```

Comment:
wait a second.... nothing is using this function. I'll get rid of it in a follow up.

But since `CreatePods` is going away, it makes sense to rename `CreatePodsWithControllerRef` to simply `CreatePods`. Then, `CreatePodsWithControllerRefAndPrefix` can simply be `CreatePodsWithPrefix`. But I can do that in the follow up too.